In [1]:
from pathlib import Path
from keyword_search_evaluation import get_scores

/home/adamp/school/cmpt419/layman-scientific-embeddings/venv/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
MODELS_PATH = Path("models")

In [3]:
def print_model_scores(model_path: str | Path):
    scores = get_scores(model_path)
    print(model_path)
    print(f"Full match score: {scores["perfect_match_score"] * 100:.2f}%")
    print(f"Related keywords score: {scores["related_keyword_score"] * 100:.2f}%")

In [4]:
for model_path in MODELS_PATH.iterdir():
    print_model_scores(model_path)
    print("")

You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


Loading weights: 100%|██████████| 199/199 [00:00<00:00, 966.46it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/vanilla-finetuned
Full match score: 43.68%
Related keywords score: 5.86%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 928.27it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/jargon-layman-finetuned-0.50
Full match score: 46.05%
Related keywords score: 6.05%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 984.38it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/jargon-layman-finetuned-0.75
Full match score: 45.55%
Related keywords score: 6.06%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 984.97it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/jargon-layman-finetuned-1.00
Full match score: 45.95%
Related keywords score: 6.12%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 976.07it/s, Materializing param=pooler.dense.weight]                               


models/jargon-layman-finetuned-0.25
Full match score: 45.65%
Related keywords score: 5.99%



In [5]:
print_model_scores("allenai/scibert_scivocab_uncased")

No sentence-transformers model found with name allenai/scibert_scivocab_uncased. Creating a new one with mean pooling.
Loading weights: 100%|██████████| 199/199 [00:00<00:00, 1189.92it/s, Materializing param=pooler.dense.weight]                               
BertModel LOAD REPORT from: allenai/scibert_scivocab_uncased
Key                                        | Status     |  | 
-------------------------------------------+------------+--+-
cls.predictions.transform.LayerNorm.bias   | UNEXPECTED |  | 
cls.seq_relationship.bias                  | UNEXPECTED |  | 
cls.predictions.decoder.weight             | UNEXPECTED |  | 
cls.predictions.transform.dense.weight     | UNEXPECTED |  | 
cls.predictions.bias                       | UNEXPECTED |  | 
cls.predictions.transform.dense.bias       | UNEXPECTED |  | 
cls.seq_relationship.weight                | UNEXPECTED |  | 
cls.predictions.decoder.bias               | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.weight | UNEXPECTED |  

allenai/scibert_scivocab_uncased
Full match score: 11.46%
Related keywords score: 1.94%


# Results

| Model | Exact keyword match score | Related keyword score |
|---|---|---|
| allenai/scibert_scivocab_uncased | 11.46% | 1.94% |
| vanilla-finetuned | **43.68%** | **5.86%** |
| jargon-layman-finetuned | <u>45.95%</u> | <u>6.12%</u> |